qa/entitlements/build.gradle

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+apply plugin: 'elasticsearch.base-internal-es-plugin'
+apply plugin: 'elasticsearch.internal-java-rest-test'
+// Necessary to use tests in Serverless
+apply plugin: 'elasticsearch.internal-test-artifact'
+
+esplugin {
+  name 'entitlement-qa'
+  description 'A test module that triggers entitlement checks'
+  classname 'org.elasticsearch.test.entitlements.EntitlementsCheckPlugin'
+}
+
+configurations {
+  entitlementBridge {
+    canBeConsumed = false
+  }
+}
+
+dependencies {
+  clusterPlugins project(':qa:entitlements')
+  entitlementBridge project(':libs:entitlement:bridge')
+}
+
+tasks.named('javaRestTest') {
+  systemProperty "tests.entitlement-bridge.jar-name", configurations.entitlementBridge.singleFile.getName()
```

Comment:
@mark-vieira I needed the name of the bridge jar, to pass down as a jvmArg to the cluster (see below).
In production, we do this in the CLI (finding the only jar in the `lib/entitlement-bridge/` subdirectory), but in the test we launch the ES cluster directly.

So I followed the guide here https://docs.gradle.org/current/userguide/cross_project_publications.html and used a pair of configurations to share the artifact name, pass it to the test as a syspro, and pass that down to the cluster via a jvmArg.
Does it makes sense to you? 
Is there a better way?

Reply:
We certainly can and should make this better. Mainly because this is eagerly resolve (read: "building") this jar when configuring this task. I'll open a follow up to improve this.

Reply:
Looking at this again I don't see why this is required. We are starting a test cluster, which is already passing the `--patch-module` CLI argument because we are using the same CLI entry points here as we do in production. We should only have to pass the `es.entitlements.enabled` to get your desired behavior.

Reply:
I've opened https://github.com/elastic/elasticsearch/pull/116779 to simplify things here. It turns out there were some changes needed to the test clusters infra to ensure that system properties are passed to the CLI as well.

Reply:
Yes, that was the reason I had to pass the args explicitly: I noticed that they were not passed to the test cluster. So the whole configurations stuff was to find the jar filename to append to the args, which now we don't need anymore.
Thanks for following up!